src/mlpack/methods/nmf/nmf_main.cpp

```diff
@@ -95,38 +95,64 @@ PARAM_STRING_IN("update_rules", "Update rules for each iteration; ( multdist | "
 PARAM_MATRIX_IN("initial_w", "Initial W matrix.", "p");
 PARAM_MATRIX_IN("initial_h", "Initial H matrix.", "q");
 
-void LoadInitialWH(const bool bindingTransposed, arma::mat& w, arma::mat& h)
+template<typename UpdateRuleType>
+void ApplyFactorization(const arma::mat& V,
+                        const size_t r,
+                        arma::mat& W,
+                        arma::mat& H)
 {
-  // Note that these datasets will typically be transposed on load, since we are
-  // likely receiving it from a row-major language, but we get it in a
-  // column-major form.  Therefore, we're actually decomposing V^T = W^T * H^T.
-  // Effectively this means we are solving, for the user, V = H*W.  Therefore,
-  // we actually have to switch what we are saving, so we will save the W we get
-  // from amf.Apply() as H, and vice versa.
-  if (bindingTransposed)
```

Comment:
We definitely can't just dump this code.  Like this comment says, the behavior we need to perform is different depending on the language we are using.

Reply:
Thanks for pointing this out! I will implement this as soon as possible.